helm/resource_release.go

```diff
@@ -1508,3 +1513,20 @@ func valuesKnown(d *schema.ResourceDiff) bool {
 	}
 	return true
 }
+
+func useChartVersion(chart string, repo string) bool {
```

Comment:
It's not really clear to me how the case of the URL pointing to an archive file is being handled here. Can you elaborate on that logic a bit?

Reply:
Sure thing, a config example would be this:
```hcl
resource "helm_release" "example" {
  name  = "redis"
  chart = "https://charts.bitnami.com/bitnami/redis-10.7.16.tgz"
}
```

within `useChartVersion` we check if the chart is a URL archive or not with `url.ParseRequestURI`. It returns nil if its a working URL. When `useChartVersion` returns true,  provider will be using the chart version found directly from the `chart / repository` provided whether that's a local directory or URL archive and will ignore the `.version` field. 

Before this PR the `.version` field would get included in the state even if it doesn't match with the chart that is used directly. This would cause a lot of confusion within the state file, this solution would be to ignore the use of `.version` when it's not needed such as local charts and URL archives.

tldr; if the chart is not coming from a repository or OCI where we may need to add which version we want then the `.version` attribute will be ignored.